homeassistant/components/tplink/__init__.py

```diff
@@ -87,7 +111,40 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up TPLink from a config entry."""
     host = entry.data[CONF_HOST]
     try:
-        device: SmartDevice = await Discover.discover_single(host, timeout=10)
+        credentials = await get_stored_credentials(hass)
+        connection_params = None
+
+        try:
+            if connection_params_dict := entry.data.get(CONF_CONNECTION_PARAMS):
```

Comment:
Please move this line out of the try block since it isn't expected to raise `SmartDeviceException`

Reply:
ok

Reply:
Done